torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1189,17 +1218,43 @@ TensorExprKernel::TensorExprKernel(const Graph& subgraph) {
         }
       }
     }
+    if (hasRandom_ && hasBroadcast_) {
+      throw std::runtime_error(
+          "Cannot support broadcast and random within one kernel");
+    }
   }
 
   // Move output operands from `tensors_` to `tensor_outputs_`
-  for (const auto& output : subgraph.outputs()) {
+  for (const auto& output : graph_->outputs()) {
     CHECK(tensors_.count(output->unique())) << "Output must be a tensor";
     tensor_outputs_.emplace_back(tensors_.at(output->unique()));
     tensors_.erase(output->unique());
   }
 }
 
+TensorExprKernel::TensorExprKernel(const std::shared_ptr<Graph>& subgraph)
+    : graph_(subgraph), code_(subgraph) {
+  try {
+    compile();
+  } catch (...) {
```

Comment:
We probably need to catch a specific exception, not everything.

Reply:
In general I agree but I'm following the current fuser on this specific point - if anything goes wrong during compilation I want to be able to hand the reins to the interpreter.